device_backends/NumericAddressedBackend/src/NumericAddressedBackend.cc

```diff
@@ -118,15 +119,29 @@ namespace ChimeraTK {
             "Register " + registerPathName + " does not support AccessMode::wait_for_new_data.");
       }
 
-      auto interruptDispatcher =
-          _interruptDispatchers.at({registerInfo.interruptCtrlNumber, registerInfo.interruptNumber});
+      auto getNestedInterruptDispatcher =
+          [](std::vector<uint32_t> interruptID,
+              std::map<uint32_t, boost::shared_ptr<NumericAddressedInterruptDispatcher>> dispatchers,
+              [[maybe_unused]] auto&& _getNestedInterruptDispatcher)
+          -> boost::shared_ptr<NumericAddressedInterruptDispatcher> {
+        auto dispatcher = dispatchers[interruptID.front()];
+        if(interruptID.size() == 1) {
+          return dispatcher;
+        }
+        throw ChimeraTK::logic_error("Nested interrupts are not supported yet!");
```

Comment:
Why do you introduce this lambda if it anyway does not yet recurse?

Reply:
Yes, you are right. Forget about this part. This has already been refactored in the real implementation.